pkg/acsengine/engine.go

```diff
@@ -135,26 +139,49 @@ func (t *TemplateGenerator) GenerateTemplate(containerService *api.ContainerServ
 	for _, file := range files {
 		bytes, e := Asset(file)
 		if e != nil {
-			return "", "", certsGenerated, fmt.Errorf("Error reading file %s, Error: %s", file, e.Error())
+			err = fmt.Errorf("Error reading file %s, Error: %s", file, e.Error())
+			return templateRaw, parametersRaw, certsGenerated, err
 		}
 		if _, err = templ.New(file).Parse(string(bytes)); err != nil {
-			return "", "", certsGenerated, err
+			return templateRaw, parametersRaw, certsGenerated, err
 		}
 	}
+	// template generation may have panics in the called functions.  This catches those panics
+	// and ensures the panic is returned as an error
+	defer func() {
```

Comment:
Do we have bugs filed for those as well?

I can't totally visualize the effects of this PR without running it in a crashy scenario and observing the difference, so please humor me:

* Before: the panic would happen, we blow up, get a stack trace (and the error?)
* After: panic is suppressed, we still end up fatally exitting, but we see just the err string instead?

Is this right? Does this add extra information, or is it just to hide the scary looking panic stack trace.